### PR TITLE
Feature/ele 262 create planning grid based on sketches

### PR DESCRIPTION
**Pull request by dannelundqvist**
Added planning overview week grid. Refactored view header and planning overview structure into multiple separate components. Added SWR for caching data fetching based on date interval. Also added som utility functions to be able to handle and convert between utc and local dates. (Search in utc, display in local timezone). Added a registry context in order to store things like locale and timeZone. Leaving actual correctness in search and sorting aside for now.

### CODE REVIEW

File: src/views/PlanningOverview/PlanningHeader/index.tsx
Comment (gularsson) [inline src/views/PlanningOverview/PlanningHeader/index.tsx:1]: Should this be named `PlanningHeader/PlanningHeader.tsx`?

File: src/components/View/View.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { type PropsWithChildren } from 'react'
+
+export const View = ({ children }: PropsWithChildren): JSX.Element => {
+  return (
+    <main className='px-3 h-full'>
+      {children}
+    </main>
```
[the remark this answers is not in the document]
Comment (dannelundqvist) [reply src/components/View/View.tsx:7]: View.tsx had more to it when it was initially created. It seems to be unnecessary now. Maybe I should remove it completely?

[the remark this answers is not in the document]
Comment (dannelundqvist) [reply src/components/View/View.tsx:7]: I removed it.

File: src/views/index.tsx [hunk 2]
```diff
@@ -1,2 +1,2 @@
 export { Editor } from './Editor'
-export { PlanningOverview } from './PlanningOverview'
+export { PlanningOverview } from './PlanningOverview/PlanningOverview'
```
[the remark this answers is not in the document]
Comment (dannelundqvist) [reply src/views/index.tsx:2]: As per discussion it should be named index.tsx. My bad for not renaming it before commit/push.

File: src/views/PlanningOverview/PlanningGrid.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    const date = localStart.substring(0, 10)
+    if (!Array.isArray(plannings[date])) {
+      plannings[date] = []
+    }
+
+    plannings[date].push(item)
+  }
+
+  return plannings
+}
```
[the remark this answers is not in the document]
Comment (dannelundqvist) [reply src/views/PlanningOverview/PlanningGrid.tsx:119]: Yes, and I suspect it is not even correct as it is now.

File: src/views/PlanningOverview/PlanningHeader/index.tsx
Comment (dannelundqvist) [reply src/views/PlanningOverview/PlanningHeader/index.tsx:1]: I think we should revisit this discussion again.

File: src/views/PlanningOverview/PlanningGridColumn.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            timeZone
+          )
+
+          return <div key={item._id} className="px-3 pb-8">
+            <div className="flex text-sm -ml-3">
+              <div className="flex-none">
+                <StatusIndicator internal={internal} />
+              </div>
+
+              <div className="flex-grow font-bold line-clamp-3">{title}</div>
```
[the remark this answers is not in the document]
Comment (gularsson) [reply src/views/PlanningOverview/PlanningGridColumn.tsx:45]: Ok, I used font-medium, seems to have a font-weight of 500...

**Review by gularsson — APPROVED** [review]

**Pull request merged**